docs/spec_draft.md

```diff
@@ -1280,6 +1281,58 @@ For boolean element type, the behavior is same as [stablehlo.or](#stablehloor).
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.map
+
+### Semantics
+
+Performs a map function `computation` over the `inputs` along the `dimensions`
+and produces a `result` tensor where each element is the result of `computation`
+applied to the corresponding elements in `inputs`.
+
+More formally, `result[i0, ..., iR-1] = computation(inputs0[i0, ..., iR-1], `
+`..., inputsN-1[i0, ..., iR-1])`.
+
+### Inputs
+
+| Name          | Type                                             |
+|---------------|--------------------------------------------------|
+| `inputs`      | variadic number of tensors of any supported type |
+| `dimensions`  | 1-dimensional tensor constant of type `si64`     |
+| `computation` | `function`                                       |
+
+### Outputs
+
+| Name     | Type                         |
+|----------|------------------------------|
+| `result` | tensor of any supported type |
+
+### Constraints
+
+  * (C1) `inputs` and `result` have the same shape.
+  * (C2) size(`inputs`) $=$ N $\ge$ 1.
+  * (C3) size(`dimensions`) $=$ rank(`inputs[k]`) for all `k` $\in$ [0, N).
+  * (C4) `dimensions[i]` $\lt$ `dimensions[i+1]` for all `i` $\in$ [0, `R`-2].
+  * (C5) `computation` has type `(tensor<E0>, ..., tensor<EN-1>) -> tensor<E'>`
+    where `Ek` $=$ element_type(`inputs[k]`) and `E'` $=$
+    element_type(`result`).
```

Comment:
From what I can see, [shape_inference.cc](https://github.com/tensorflow/tensorflow/blob/1d34295f45965c7090039773dea44c25ac00ec00/tensorflow/compiler/xla/service/shape_inference.cc#L1172-L1181) allows some limited form of broadcasting and mixed-type precision. Let's mark Verification as revisit and leave figuring this out to #442 and #369.

Reply:
Thanks for linking the related issue. Verification is already marked `revisit` as mentioned in my comment above.